### PR TITLE
Support Phpunit 6 without impacting app runtime

### PR DESCRIPTION
**Pull request by antograssiot**
Another attempt to support PHPUnit without forcing file loading for every request.

I tested it with composer installation and .phar for PHPUnit 5.7 and 6 but I would appreciate if one can quickly check one of is app using this branch to confirm it does work properly before it (cc @dereuromark )

I added a "conflict" definition in the `composer.json` to alert composer users.

Finally, what fo you think of adding the following code inside the ifs:
```
if (version_compare(\PHPUnit_Runner_Version::id(), '5.7', '<')) {
        trigger_error(sprintf('Your PHPUnit Version must be at least 5.7.0 to use CakePHP Testsuite, found %s', \PHPUnit_Runner_Version::id()), E_USER_ERROR);
    }
```

I think it can help people to better understand what's going on than getting an standard error that could looks like 
```
.../cakephp/src/TestSuite/Fixture/FixtureInjector.php on line 38
````

### CODE REVIEW

**Comment by markstory** [comment]
I think the proposed warning makes sense :+1:

**Comment by antograssiot** [comment]
Ok I've added the warning and hopefully fix the issue users have.

@markstory Please don't merge this just yet, I'll try to get feedback from #10260

File: src/TestSuite/Fixture/FixtureInjector.php [hunk 1]
```diff
@@ -14,6 +14,21 @@
  */
 namespace Cake\TestSuite\Fixture;
 
+if (class_exists('PHPUnit_Runner_Version')) {
+    if (version_compare(\PHPUnit_Runner_Version::id(), '5.7', '<')) {
+        trigger_error(sprintf('Your PHPUnit Version must be at least 5.7.0 to use CakePHP Testsuite, found %s', \PHPUnit_Runner_Version::id()), E_USER_ERROR);
+    }
+    class_alias('PHPUnit_Framework_Test', 'PHPUnit\Framework\Test');
+    class_alias('PHPUnit_Framework_Warning', 'PHPUnit\Framework\Warning');
```
Comment (markstory) [inline src/TestSuite/Fixture/FixtureInjector.php]: Why don't these have `class_exists()` calls?

File: src/TestSuite/Fixture/FixtureInjector.php [hunk 2]
```diff
@@ -14,6 +14,21 @@
...
+
+    if (!class_exists('PHPUnit\Framework\TestSuite')) {
+        class_alias('PHPUnit_Framework_TestSuite', 'PHPUnit\Framework\TestSuite');
+    }
+    if (class_exists('PHPUnit_Runner_Version') && !class_exists('PHPUnit\Framework\AssertionFailedError')) {
```
Comment (markstory) [inline src/TestSuite/Fixture/FixtureInjector.php]: Why does one depend on the other?

File: src/TestSuite/Fixture/FixtureInjector.php (hunk 1 above)
Comment (antograssiot) [inline src/TestSuite/Fixture/FixtureInjector.php]: @markstory my reasoning was the following:
If a old PHPUNIT version class is found, I need de define aliases
If I need to define the same aliases in several places, I check if it was not done already.

Comment (antograssiot) [inline src/TestSuite/Fixture/FixtureInjector.php]: Using the `PHPUnit_Framework_Version` class also determines if PHPUnit <6 is present at all and only defines those aliases in this case

File: src/TestSuite/Fixture/FixtureInjector.php (hunk 2 above)
Comment (antograssiot) [inline src/TestSuite/Fixture/FixtureInjector.php]: I can remove the first check it is already nested in

**Comment by lorenzo** [comment]
Great job @antograssiot

**Pull request merged**